### PR TITLE
avoid dangling keepalive session when setup fails

### LINKED ISSUE #15: Keepalive session isn't cleaned up in case of failure
**Describe the bug**
In case of a setup failure the keepalive session isn't cleaned up.

**To Reproduce**
1. Run the UAT pointing the `PATH` to a directory not containing purebred, e.g. `PATH=foo cabal --enable-nix test --show-details=streaming uat`
2. First run fails with `Exception: purebred: startProcess: runInteractiveProcess: exec: does not exist (No such file or directory)`
3. Now run it again and the *first* exception is:
```
user acceptance tests                                                                                                                 
  user can view mail:                                       duplicate session: keepalive
```

**Expected behavior**
Keepalive session is cleaned up in case of failure.

**Additional context**
None

### PR DESCRIPTION
**Pull request by frasertweedale**
Fixes: https://github.com/purebred-mua/tasty-tmux/issues/15

### CODE REVIEW

**Comment by frasertweedale** [comment]
@romanofski does this solve the issue for you?

File: src/Test/Tasty/Tmux.hs [hunk 1]
```diff
@@ -292,12 +292,13 @@ testTmux
   -> [TestCase a]
   -> TestTree
 testTmux pre post tests =
-  withResource (frameworkPre *> pre) (\a -> post a *> frameworkPost) $ \env ->
-    testGroup "user acceptance tests" $ zipWith ($ env) tests [0..]
+  withResource frameworkPre frameworkPost $ \ioKeepalive ->
+    withResource (ioKeepalive *> pre) post $ \env ->
+      testGroup "user acceptance tests" $ zipWith ($ env) tests [0..]
```
Comment (romanofski) [inline src/Test/Tasty/Tmux.hs:297]: This looks good! Thank you! I'll give it a try tonight :))

**Comment by romanofski** [comment]
Coolio! Works as expected. Thanks @frasertweedale

**Comment by frasertweedale** [comment]
@romanofski I'll cut a release later this week (maybe tonight).

**Pull request merged**